### PR TITLE
node-neigh: Fix concurrent arping update unit test flake

### LINKED ISSUE #16075: CI: linuxPrivilegedIPv4OnlyTestSuite.SetUpTest
## CI failure
Output from failure:
```
17:05:08  	 
17:05:08  	 START: node_linux_test.go:955: linuxPrivilegedIPv4OnlyTestSuite.TestArpPingHandling
17:05:08  	 START: node_linux_test.go:116: linuxPrivilegedIPv4OnlyTestSuite.SetUpTest
17:05:08  	 PASS: node_linux_test.go:116: linuxPrivilegedIPv4OnlyTestSuite.SetUpTest	0.001s
17:05:08  	 
17:05:08  	 node_linux_test.go:1161:
17:05:08  	     c.Assert(neighHwAddr(ip1.String()), check.Equals, mac.String())
17:05:08  	 ... obtained string = "f6:27:bb:80:88:10"
17:05:08  	 ... expected string = "4a:4c:8c:84:d1:20"
17:05:08  	 
```

Link to CI build: https://jenkins.cilium.io/job/Cilium-PR-Runtime-4.9/4628/execution/node/99/log/?consoleFull

Link to PR where this happened: https://github.com/cilium/cilium/pull/16041

### PR DESCRIPTION
**Pull request by brb**
See commit msgs.

Fix #16075

### CODE REVIEW

**Comment by brb** [comment]
test-runtime

**Comment by brb** [comment]
test-runtime

File: pkg/datapath/linux/node.go [hunk 1]
```diff
@@ -788,7 +788,7 @@ func (n *linuxNodeHandler) insertNeighbor(ctx context.Context, newNode *nodeType
 			// issued arping after us, as it might have a more recent hwAddr value.
 			return
 		}
-		n.neighLastPingByNextHop[nextHopStr] = time.Now()
+		n.neighLastPingByNextHop[nextHopStr] = now
```
Comment (nebril) [inline pkg/datapath/linux/node.go:791]: If line 771 evaluates to `false`, this will be be an empty time, won't it?

**Review by nebril — CHANGES_REQUESTED** [review]

File: pkg/datapath/linux/node.go (hunk 1 above)
Comment (brb) [reply pkg/datapath/linux/node.go:791]: Yeah, but then `hwAddr` will be nil too, so this won't be evaluated.

Comment (nebril) [reply pkg/datapath/linux/node.go:791]: Tru dat!

**Review by nebril — APPROVED** [review]

**Review by jrfastab — APPROVED** [review]
lgtm

**Pull request merged**